Runtime/Utils/CDRSerializer.cs

```diff
@@ -0,0 +1,333 @@
+            }
+        }
+
+        private static void WriteArray(BinaryWriter writer, Array array)
+        {
+            foreach (var item in array)
+            {
+                Type itemType = item.GetType();
+                if (ROS2Serialization) AlignStream(writer, GetAlignment(itemType));
+
+                if (itemType == typeof(byte))
+                {
+                    writer.Write((byte)item);
+                }
+                else if (itemType == typeof(int))
+                {
+                    writer.Write((int)item);
+                }
+                else if (itemType == typeof(uint))
+                {
+                    writer.Write((uint)item);
+                }
+                else if (itemType == typeof(float))
+                {
+                    writer.Write((float)item);
+                }
+                else if (itemType == typeof(double))
+                {
+                    writer.Write((double)item);
+                }
+                else if (itemType == typeof(string))
+                {
+                    WriteString(writer, (string)item);
+                }
+                else if (itemType.IsClass && !itemType.IsPrimitive) // Classes
+                {
+                    SerializeObject(writer, item);
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Unsupported array item type: {itemType.Name}");
+                }
+            }
+        }
+
+        private static int GetAlignment(Type type)
+        {
+            if (type == typeof(byte)) return 1;
+            if (type == typeof(int)) return 4;
+            if (type == typeof(uint)) return 4;
+            if (type == typeof(float)) return 4;
+            if (type == typeof(string)) return 4;
+            if (type == typeof(double)) return 4;
+            if (type.IsClass) return 1; // Assuming classes don't need alignment
+            throw new InvalidOperationException($"Unknown type for alignment: {type.Name}");
+        }
+
+        private static void AlignStream(BinaryWriter writer, int alignment)
+        {
+            long position = writer.BaseStream.Position;
+            long padding = (alignment - (position % alignment)) % alignment;
+            for (int i = 0; i < padding; i++)
+            {
+                writer.Write((byte)0); // Write padding bytes
+            }
+        }
+
+        public static T Deserialize<T>(byte[] data) where T : new()
+        {
+            using (MemoryStream ms = new MemoryStream(data))
+            {
+                using (BinaryReader reader = new BinaryReader(ms))
+                {
+                    reader.BaseStream.Seek(4, SeekOrigin.Begin); // Skip the first 4 header bytes
+                    return (T)DeserializeObject(reader, typeof(T));
+                }
+            }
+        }
+        
+        public static object DeserializeMsg(byte[] data)
```

Comment:
Header have json structure

Reply:
This was before I got the new updated packet structure, so I wrote this in case we were going to binary serialize everything. 

Anyway, I've removed it and will create the json header while sending the message since this class is primary focus on doing the binary serialization